devices/tuya.js

```diff
@@ -230,8 +229,26 @@ module.exports = [
             {vendor: 'Earda', model: 'EDM-1ZAA-EU'},
             {vendor: 'Earda', model: 'EDM-1ZAB-EU'},
             {vendor: 'Earda', model: 'EDM-1ZBA-EU'},
-            {vendor: 'Mercator ikuü', model: 'SISWD01'},
-        ],
+            ],
+    },
+            // Ikuu single gang no neutral dimmer device
+    {
+            fingerprint: [
+                {modelID: 'TS0601', manufacturerName: '_TZE200_swaamsoy'},
+            ],
+            model: 'SSWD01',
+            vendor: 'Mercator ikuü',
+            description: 'Zigbee smart dimmer',
+            fromZigbee: [fz.ikuu_dimmer, fz.ignore_basic_report],
```

Comment:
I'm wondering if a new definition is really needed, the only difference is `fz. ikuu_dimmer ` vs `fz.tuya_dimmer`, I think it is better to just fix the `fz.tuya_dimmer` (add the datapoint for the brightness)

Reply:
I simply don't know how the other Tuya dimmer devices work, if they all have the same data points or not.

If they are all the same, then yeah that makes more sense. I didn't want to take the risk of breaking other people's stuff, but we can do that if it's best.

Reply:
Can you try just changing:

```js
tuya_dimmer: {
    cluster: 'manuSpecificTuya',
    type: ['commandGetData', 'commandSetDataResponse'],
    convert: (model, msg, publish, options, meta) => {
        const value = tuya.getDataValue(msg.data.datatype, msg.data.data);
        if (msg.data.dp === tuya.dataPoints.state) {
            return {state: value ? 'ON': 'OFF'};
        } else if (meta.device.manufacturerName === '_TZE200_swaamsoy' && msg.data.dp === 3) {
            return {brightness: mapNumberRange(value, 10, 1000, 0, 254), level: value};
        } else { // TODO: Unknown dp, assumed value type
            return {brightness: mapNumberRange(value, 10, 1000, 0, 254), level: value};
        }
    },
},
```

Reply:
Maybe I didn't explain the issue too well.

Data point 2 is the actual dimmer brightness level.
Data point 3 is a minimum brightness level that is set. This is because the dimming switch has no neutral wire, going lower than the minimum setting in brightness results in the load LED turning.

So when we see data point 2, we get brightness ok. But when we see data point 3, that else assumes it's brightness and sets incorrect values. So that proposed change will result in the same issue if I am not mistaken?

Would need a couple more ifs there to say:
if ikuu and dp is 2 then set brightness
elif ikuu and dp is 3 then set min-brightness
elif ikuu and dp is 6 set timer
else assume brightness

Want me to do that? But seems super hacky.

Reply:
You are right, what about:

```js
tuya_dimmer: {
    cluster: 'manuSpecificTuya',
    type: ['commandGetData', 'commandSetDataResponse'],
    convert: (model, msg, publish, options, meta) => {
        const value = tuya.getDataValue(msg.data.datatype, msg.data.data);
        if (msg.data.dp === tuya.dataPoints.state) {
            return {state: value ? 'ON': 'OFF'};
        } else {
            if (meta.device.manufacturerName === '_TZE200_swaamsoy') {
                // https://github.com/Koenkk/zigbee-herdsman-converters/pull/3004
                if (msg.data.dp === 3) {
                    return {brightness: mapNumberRange(value, 10, 1000, 0, 254), level: value};
                }
            } else { // TODO: Unknown dp, assumed value type
                return {brightness: mapNumberRange(value, 10, 1000, 0, 254), level: value};
            }
        }
    },
},
```

Reply:
Ok, I have made that change with a minor adjustment. Does work in my testing.

Those other data points (3 and 6) I can't figure out yet, I will get to them when ever I can.